.github/workflows/build_release.yml

```
@@ -0,0 +1,77 @@
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          distribution: goreleaser
+          version: ~> v2
+          args: release --clean ${{ inputs.new_tag == '' && '--snapshot' || '' }}
+          workdir: server
+        env:
+          GORELEASER_CURRENT_TAG: ${{ inputs.new_tag == '' && '0.0.0' || inputs.new_tag }}
+      - name: Rename artifacts
+        if: ${{ inputs.name != '' }}
+        run: for f in $ARTIFACTS; do mv $f $(echo $f | sed -E 's/_0\.0\.0-SNAPSHOT-[^_]*/_${{ inputs.name }}/'); done
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:28: Double quote to prevent globbing and word splitting

(shellcheck)